client/ayon_resolve/plugins/publish/collect_shots.py

```diff
@@ -0,0 +1,112 @@
+        if not context.data.get("editorialSharedData"):
+            context.data["editorialSharedData"] = {}
+
+        context.data["editorialSharedData"][instance_id] = {
+            key: value for key, value in instance.data.items()
+            if key in cls.SHARED_KEYS
+        }
+
+    def process(self, instance):
+        """
+        Args:
+            instance (pyblish.Instance): The shot instance to update.
+        """
+        instance.data["integrate"] = False  # no representation for shot
+
+        # Adjust instance data from parent otio timeline.
+        otio_timeline = instance.context.data["otioTimeline"]
+        otio_clip, marker = utils.get_marker_from_clip_index(
+            otio_timeline, instance.data["clip_index"]
+        )
+        if not otio_clip:
+            raise RuntimeError("Could not retrieve otioClip for shot %r", instance)
+
+        # Compute fps from creator attribute.
+        if instance.data['creator_attributes']["fps"] == "from_selection":
+            instance.data['creator_attributes']["fps"] = instance.context.data["fps"]
+
+        # Retrieve AyonData marker for associated clip.
+        instance.data["otioClip"] = otio_clip
+        creator_id = instance.data["creator_identifier"]
+        inst_data = marker.metadata["resolve_sub_products"].get(creator_id, {})
+
+        # Overwrite settings with clip metadata is "sourceResolution"
+        overwrite_clip_metadata = inst_data.get("sourceResolution", False)
+        if overwrite_clip_metadata:
+            clip_metadata = inst_data["clip_source_resolution"]
+            width = clip_metadata["width"]
+            height = clip_metadata["height"]
+            pixel_aspect = clip_metadata["pixelAspect"]
+
+        else:
+            # AYON's OTIO export = resolution from timeline metadata.
+            # This is metadata is inserted by ayon_resolve.otio.davinci_export.
+            width = height = None
+            try:
+                width = otio_timeline.metadata["width"]
+                height = otio_timeline.metadata["height"]
+                pixel_aspect = otio_timeline.metadata["pixelAspect"]
+
+            except KeyError:
+                # Retrieve resolution for project.
+                project = lib.get_current_project()
+                project_settings = project.GetSetting()
+                try:
+                    pixel_aspect = int(project_settings["timelinePixelAspectRatio"])
+                except ValueError:
+                    pixel_aspect = 1.0
+
+                width = int(project_settings["timelineResolutionWidth"])
+                height = int(project_settings["timelineResolutionHeight"])
+
+        instance.data.update(
+            {
+                "resolutionWidth": width,
+                "resolutionHeight": height,
+                "pixelAspect": pixel_aspect,
+            }
+        )
+
+        self._inject_editorial_shared_data(instance)
+        self.log.debug(pprint.pformat(instance.data))
```

Comment:
Can we avoid this heavy verbose logging here? Or be more specific about what we want to log?
```suggestion
```

Reply:
I'm gonna let @jakubjezek001 answer this one. I've basically reported it from existing plugins for consistency.

From my experience, it's bruteforce but very valuable for a developper (hence debug) especially for troubleshooting.
But if it ends up in user face that'll be quite disturbing for an artist indeed.

Reply:
If this is the kind of debugging we need - I'd argue that we should instead focus on allowing such functionality inside the Publisher UI itself (similar to the Pyblish Debug Stepper https://github.com/ynput/ayon-core/pull/753) so that one could e.g. have the publisher itself set e.g. debug flags for certain plug-ins and then e.g. add a "profiled" report that e.g. reports the "changes" to the `instance.data` from that plug-in, etc.

We should really try and refrain from having each plug-in enforcing this massive data point all the time. It's slow - and on my old machine; it's **very slow**. The majority of my publish time is spent around these calls ;) (luckily these are without the `json.dumps` around it because those were even slower)

Reply:
That makes sense, especially if it impacts performances. I'll remove those, I guess a developer can easily bring them back if needed while troubleshooting.